src/runner.ts

```diff
@@ -228,15 +209,54 @@ export class TestsRunner {
         cliProcess.on("error", (err) => { reject(err); });
     }
 
-    private getJavaPath(javaRootPath: string): string | undefined {
-        if (!javaRootPath) {
+    private getSOAtestJavaPath(installDir: string): string | undefined {
+        let javaFilePath;
+        if (installDir && fs.existsSync(installDir)) {
+            javaFilePath = this.doGetSOAtestJavaPath(installDir);
+        } else {
+            try {
+                const soatestcliPath = which.sync('soatestcli');
+                installDir = soatestcliPath.substring(0, soatestcliPath.lastIndexOf('soatestcli') - 1);
+                javaFilePath = this.doGetSOAtestJavaPath(installDir);
+            } catch (error) {
+                core.warning(messages.soatest_install_dir_not_found);
+                javaFilePath = undefined;
+            }
+        }
+
+        if (!javaFilePath) {
+            core.warning(messages.java_not_found_in_soatest_install_dir);
+        } else {
+            core.info(messagesFormatter.format(messages.found_java_at, javaFilePath));
+        }
+        return javaFilePath;
+    }
+
+    private doGetSOAtestJavaPath(installDir: string): string | undefined {
+        core.info(messagesFormatter.format(messages.find_java_in_soatest_install_dir, installDir));
```

Comment:
I think it's better to change all messages related to Java to debug messages since it's not necessary for user to know the inner logic in a normal workflow.